data/lib/src/datasource_impl/sort_datasource_impl.dart

```diff
@@ -0,0 +1,98 @@
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:data/data.dart';
+import 'package:domain/domain.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class SortDataSourceImpl implements SortDataSource {
+  final SharedPreferences _sharedPreferences;
+
+  SortDataSourceImpl(this._sharedPreferences);
+
+  @override
+  Future<Sorter> getSorter(OrderScreen orderScreen) {
+    final orderByStr = _sharedPreferences
+            .getString('sort_file_order_by_${orderScreen.toString()}') ??
+        OrderBy.modificationDate.toString();
+    final orderTypeStr = _sharedPreferences
+            .getString('sort_file_order_type_${orderScreen.toString()}') ??
+        OrderType.descending.toString();
+
+    var orderBy = OrderBy.modificationDate;
+
+    if (orderByStr == OrderBy.modificationDate.toString()) {
+      orderBy = OrderBy.modificationDate;
+    } else if (orderByStr == OrderBy.creationDate.toString()) {
+      orderBy = OrderBy.creationDate;
+    } else if (orderByStr == OrderBy.fileSize.toString()) {
+      orderBy = OrderBy.fileSize;
+    } else if (orderByStr == OrderBy.name.toString()) {
+      orderBy = OrderBy.name;
+    } else if (orderByStr == OrderBy.shared.toString()) {
+      orderBy = OrderBy.shared;
+    }
+    final orderType = orderTypeStr == OrderType.descending.toString() ? OrderType.descending : OrderType.ascending;
+
+    return Future.sync(() async {
```

Comment:
Why not all the method should be wrapped in `async`

Reply:
@manhduc-nguyen Can you pair this with Dat? Thanks in advance

Reply:
it's  sync